core/keyboard_nav/ast_node.js

```diff
@@ -470,6 +470,23 @@ Blockly.ASTNode.prototype.findFirstFieldOrInput_ = function(block) {
   return null;
 };
 
+/**
+ * Finds the source block of the location of this node.
+ * @return {Blockly.Block} The source block of the location, or null if the node
+ * is of type workspace.
```

Comment:
Add visibility

Reply:
(unless you're trying to make this public)

Reply:
Yeah was trying to make it public. I think it is helpful for people trying to create different cursors.